dbfit-java/core/src/main/java/dbfit/util/sql/PreparedStatements.java

```diff
@@ -8,11 +8,23 @@ private static String buildParamList(int numberOfParameters) {
         return join(repeat("?", numberOfParameters), ", ");
     }
 
-    public static String buildStoredProcedureCall(String procName, int numberOfPararameters) {
-        return "{ call " + procName + "(" + buildParamList(numberOfPararameters) + ")}";
+    private static String returnValueParam(boolean returnsValue) {
+        return returnsValue ? "? = " : "";
     }
 
-    public static String buildFunctionCall(String procName, int numberOfParameters) {
-        return "{ ? = call " + procName + "(" + buildParamList(numberOfParameters - 1) + ")}";
+    private static String storedRoutineExecution(String routineName, int numberOfParameters) {
+        return "call " + routineName + "(" + buildParamList(numberOfParameters) + ")";
+    }
+
+    public static String storedRoutineStatement(
+            String routineName, int numberOfParams, boolean hasReturnValueParam) {
+        int executionParams = numberOfParams - (hasReturnValueParam ? 1 : 0);
```

Comment:
In one of the cases we're counting the return value as a parameter, in the other - not. It's good to somehow differentiate... That's not quite clear based on `args` vs `params` either - but it's kind of a hint that they're different.